### PR TITLE
Feat dashboard access

### PR DESCRIPTION
**Pull request by saurabh1920**

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -1827,6 +1943,29 @@ def sqllab_viz(self) -> FlaskResponse:  # pylint: disable=no-self-use
         table.columns = cols
         table.metrics = [SqlMetric(metric_name="count", expression="count(*)")]
         db.session.commit()
+
```
Comment (vipuls-turtlemint) [inline superset/views/core.py:1962]: Why this code is required, this is commented.

File: superset/views/utils.py [hunk 2]
```diff
@@ -379,6 +379,13 @@ def is_owner(obj: Union[Dashboard, Slice], user: User) -> bool:
     return obj and user in obj.owners
 
 
+def check_datasource_perms_extended(
```
Comment (vipuls-turtlemint) [inline superset/views/utils.py:382]: put a comment here explaining what this function trying to do.

File: superset/views/core.py [hunk 3]
```diff
@@ -1570,6 +1585,104 @@ def publish(  # pylint: disable=no-self-use
...
+        return True
+
+    def is_owner(self, dash):
+        user_id = g.user.id
+        owners_id = [owner.id for owner in dash.owners]
+        is_owner = True if user_id in owners_id else False
+        return is_owner
+
+    @expose("/dashboard/set_roles/", methods=["POST"])
+    def set_dashboard_roles(self) -> FlaskResponse:
```
Comment (vipuls-turtlemint) [inline superset/views/core.py:1617]: Put a comment on each of the new function we have added to enable Dashboard sharing

File: superset/views/core.py (hunk 1 above)
Comment (saurabh1920) [reply superset/views/core.py:1962]: Done

File: superset/views/core.py (hunk 3 above)
Comment (saurabh1920) [reply superset/views/core.py:1617]: Done

File: superset/views/utils.py (hunk 2 above)
Comment (saurabh1920) [reply superset/views/utils.py:382]: Done

**Pull request merged**